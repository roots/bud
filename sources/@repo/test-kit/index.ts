import type {Bud, Context} from '@roots/bud-framework'

import {path} from '@repo/constants'
import {factory} from '@roots/bud'

export type Options = {
  extensions?: Array<string>
} & Omit<Partial<Context>, `extensions`>

const makeTestBud = async (options: Options = {}): Promise<Bud> =>
  await factory({
    basedir: options.basedir ?? path(`tests`, `util`, `project`),
    cache: false,
    dry: true,
    force: true,
    mode: `production`,
    notify: false,
    silent: true,
    ...options,
  })

export type {Bud}
export {makeTestBud as factory, makeTestBud}
