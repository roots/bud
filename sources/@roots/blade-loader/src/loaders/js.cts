export default function (source: string) {
  return [...source.matchAll(/@js(?<content>[\s\S]*?)@endjs/g)]
    .map(match => match.groups.content)
    .filter(Boolean)
    .join(`\n`)
}
