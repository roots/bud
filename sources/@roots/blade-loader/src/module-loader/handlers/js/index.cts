import {extract} from '@roots/blade-loader/module-loader/handlers'

export const extension = `js`
export const pattern =
  /@module\(['"]js['"]\)(?<content>[\s\S]*?)@endmodule/g

export default (source: string) => extract(source, pattern)
