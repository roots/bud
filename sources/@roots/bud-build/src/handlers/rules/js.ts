import type {Factory} from '../index.js'

export const js: Factory = ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setUse([])
