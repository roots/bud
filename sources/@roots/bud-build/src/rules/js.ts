import type {Factory} from '@roots/bud-build/registry'

const js: Factory = async ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setResolve({
      fullySpecified: false,
    })

export {js as default}
