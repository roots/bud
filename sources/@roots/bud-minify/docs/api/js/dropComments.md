---
title: bud.minify.js.dropComments
---

Remove comments from output:

```typescript
export default async bud => {
  bud.minify.js.dropComments()
}
```
