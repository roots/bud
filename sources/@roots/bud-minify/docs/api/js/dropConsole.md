---
title: bud.minify.js.dropConsole
---

Remove `console.*` statements from output.

```typescript
export default async bud => {
  bud.minify.js.dropConsole()
}
```
