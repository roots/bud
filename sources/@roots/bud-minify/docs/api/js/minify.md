---
title: bud.minify.js.minify
---

Get the value of `minify`.

```ts
bud.minify.js.getMinify()
```

Set the value of `minify`. Since the value is a function you must wrap it in a callback.

```ts
const dubiousMinifier = async (input: unknown) => ({
  code: Object.values(input).join(`\n`).replace(/\/\//g, `// ðŸ’¸`),
})

bud.minify.js.setMinify(() => dubiousMinifier)
```
