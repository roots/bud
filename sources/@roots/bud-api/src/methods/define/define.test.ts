import {Bud, factory} from '@repo/test-kit'
import {beforeEach, describe, expect, it} from 'vitest'

import {define as defineMethod} from './index.js'

describe(`bud.define`, function () {
  let bud: Bud
  let define: defineMethod

  beforeEach(async () => {
    bud = await factory()
    define = defineMethod.bind(bud)
  })

  it(`should be a function`, () => {
    expect(define).toBeInstanceOf(Function)
  })

  it(`should return bud`, () => {
    const returned = define({})
    expect(returned).toBe(bud)
  })

  it(`adds definitions`, () => {
    define({DEFINED_KEY: `DEFINED_VALUE`})

    expect(
      bud.extensions.get(`@roots/bud-extensions/webpack-define-plugin`)
        .options.DEFINED_KEY,
    ).toEqual(`DEFINED_VALUE`)
  })

  it(`adds PUBLIC_APP_TITLE from env`, async () => {
    await bud.run()
    expect(
      bud.extensions.get(`@roots/bud-extensions/webpack-define-plugin`)
        .options.APP_TITLE,
    ).toBeDefined()
  })

  it(`matches snapshot`, () => {
    expect(
      bud.extensions.get(`@roots/bud-extensions/webpack-define-plugin`)
        .options,
    ).toMatchSnapshot()
  })
})
