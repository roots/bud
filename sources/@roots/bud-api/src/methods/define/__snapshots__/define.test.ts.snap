// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bud.define > matches snapshot 1`] = `
{
  "APP_DESCRIPTION": "\\"test app description\\"",
  "APP_TITLE": "\\"bud.js test app\\"",
}
`;
